src/main/java/com/gard/testbed/activitySystem/IState.java

```diff
@@ -3,6 +3,19 @@
 import java.util.List;
 
 public interface IState {
+
+    /**
+     * Returns true if all outgoing transition triggers test true against blackboard
+     * @return
+     */
+    boolean isCompleted(IBlackboard player);
```

Comment:
ok - I was working on covering concurrent as well - I'm cautious about not having a solid picture of the final mechanics and going off in the wrong direction - could we have a quick chat when you have the time?